doc/api/http.md

```diff
@@ -362,8 +362,8 @@ const proxy = http.createServer((req, res) => {
 });
 proxy.on('connect', (req, cltSocket, head) => {
   // Connect to an origin server
-  const srvUrl = url.parse(`http://${req.url}`);
-  const srvSocket = net.connect(srvUrl.port, srvUrl.hostname, () => {
+  const { port, hostname } = new URL(`http://${req.url}`);
+  const srvSocket = net.connect(port || 80, hostname, () => {
     cltSocket.write('HTTP/1.1 200 Connection Established\r\n' +
                     'Proxy-agent: Node.js-Proxy\r\n' +
```

Comment:
For good measure, this is not a valid HTTP header and it might be good to just remove it here.

Reply:
Would it make sense to change it to `User-agent`?

Reply:
Oh, wait, never mind, not the client. Ignore.

Reply:
In 2012, [RFC 6648](https://tools.ietf.org/html/rfc6648) deprecated the use of the `X-` prefix for custom HTTP headers. Instead, there's now a set of guidelines on how to name non-spec'd HTTP headers. As such, there's to my knowledge therefore no longer such a thing as "invalid" HTTP headers.

Reply:
I'm going to land this one now as it's not really about the name of this header. If we feel like it's better to rename it, we can always just land it in a separate PR 🙂